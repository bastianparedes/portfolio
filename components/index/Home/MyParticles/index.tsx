import React, { useCallback } from 'react';
import Particles from 'react-tsparticles';
import { loadFull } from 'tsparticles';
import type { Engine } from 'tsparticles-engine';
import styles from './styles.module.scss';

const MyParticles = (): JSX.Element => {
  const particlesInit = useCallback(async (engine: Engine) => {
    await loadFull(engine);
  }, []);

  return (
    <Particles
      className={styles.particles}
      init={particlesInit}
      options={{
        autoPlay: true,
        fullScreen: false,
        fpsLimit: 60,
        interactivity: {
          detectsOn: 'canvas',
          events: {
            onHover: {
              enable: true,
              mode: 'grab'
            },
            onClick: {
              enable: true,
              mode: 'push'
            }
          },
          modes: {
            push: {
              quantity: 1
            },
            grab: {
              distance: 200
            }
          }
        },
        background: {},
        particles: {
          color: {
            value: '#2034fa'
          },
          number: {
            value: 30,
            limit: 30
          },
          move: {
            enable: true,
            direction: 'top',
            random: true,
            speed: 2,
            trail: {
              enable: false
            },
            vibrate: false
          },
          size: {
            value: {
              min: 1,
              max: 5
            },
            animation: {
              enable: true,
              speed: 5
            }
          },
          shape: {
            type: 'circle'
          },
          links: {
            color: '#2034fa',
            distance: 200,
            enable: true,
            width: 2
          }
        },
        pauseOnBlur: true,
        pauseOnOutsideViewport: true
      }}
    />
  );
};

export default MyParticles;
