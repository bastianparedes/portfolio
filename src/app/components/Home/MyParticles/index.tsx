import React, { useCallback } from 'react';

import Particles from 'react-tsparticles';
import { loadFull } from 'tsparticles';
import type { Engine } from 'tsparticles-engine';

import styles from './styles.module.scss';

const MyParticles = () => {
  const particlesInit = useCallback(async (engine: Engine) => {
    await loadFull(engine as any);
  }, []);

  return (
    <Particles
      className={styles.particles}
      init={particlesInit}
      options={{
        autoPlay: true,
        background: {},
        fpsLimit: 60,
        fullScreen: false,
        interactivity: {
          detectsOn: 'canvas',
          events: {
            onClick: {
              enable: true,
              mode: 'push'
            },
            onHover: {
              enable: true,
              mode: 'grab'
            }
          },
          modes: {
            grab: {
              distance: 200
            },
            push: {
              quantity: 1
            }
          }
        },
        particles: {
          color: {
            value: '#2034fa'
          },
          links: {
            color: '#2034fa',
            distance: 200,
            enable: true,
            width: 2
          },
          move: {
            direction: 'top',
            enable: true,
            random: true,
            speed: 2,
            trail: {
              enable: false
            },
            vibrate: false
          },
          number: {
            limit: 30,
            value: 30
          },
          shape: {
            type: 'circle'
          },
          size: {
            animation: {
              enable: true,
              speed: 5
            },
            value: {
              max: 5,
              min: 1
            }
          }
        },
        pauseOnBlur: true,
        pauseOnOutsideViewport: true
      }}
    />
  );
};

export default MyParticles;
