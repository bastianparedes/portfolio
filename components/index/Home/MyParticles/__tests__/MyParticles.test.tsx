import React from 'react';
import MyParticles from '..';
import { render } from '@testing-library/react';

interface typeInit {
  init: () => void;
}

jest.mock('react-tsparticles', () => (init: typeInit) => {
  init.init();
  return <></>;
});
jest.mock('tsparticles', () => ({
  loadFull: () => null
}));

describe('<MyParticles />', () => {
  it('should render', () => {
    const { container } = render(<MyParticles />);
    expect(container).toMatchSnapshot();
  });
});
