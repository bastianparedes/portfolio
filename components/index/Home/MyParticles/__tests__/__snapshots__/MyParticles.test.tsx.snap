// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyParticles /> should render 1`] = `<div />`;
